### ISSUE TITLE
How to create 2 items at once where each require other in relation?

### ISSUE THREAD

**Issue by pie6k** [prisma/prisma-client-js#887]
## Problem
Imagine such model:

```
model Owner {
  id   String @id @default(cuid())
  name String
  car  Car
}

model Car {
  id      String @id @default(cuid())
  name    String
  owner   Owner  @relation(fields: [ownerId], references: [id])
  ownerId String
}
```

Car has an owner and each owner must have a car. Both sides of the relation are required all the time.

I struggle with creating such items.

If I try to create Owner, I have to pass Car to be connected to it. But as car is not yet created, how do I know what ID to pass? If I want to create Car first - I have the same problem as I need id of the owner to connect it with

```ts
db.car.create({
  data: {
    name: '911',
    owner: {
      create: {
        name: 'Bob',
        // Car is required
        car: {
          // What do I put here as car is not created yet?
        }
      }
    }
  }
})
```

I was trying to use `connectOrCreate` as it seemed to not require 2nd side of the relation when connecting, but it didn't work.

```
`Error for binding '3': AssertionError("[Query Graph] Expected a valid parent ID to be present for a nested connect on a one-to-many relation.")`
```

In other words - those both items have to be created at once and connected instantly.

**Comment by pantharshit00** [comment]
@pie6k Sorry for the delay here, this one got slipped in my notifications.

So `create` will also connect to record you are creating, no need to provide it. If you use Typescript here, you will also get a warning like so:
![image](https://user-images.githubusercontent.com/22195362/95980338-a55a9080-0e3a-11eb-93b9-d8e5a88c1578.png)

This call should suffice for creating a `Car` and also creating an owner for it:
```
  const data = await prisma.car.create({
    data: {
      name: "911",
      owner: {
        create: {
          name: "Bob",
        },
      },
    },
  });

```


Prisma run the following SQL for this to ensure they are also connected:
```sql
BEGIN
INSERT INTO "public"."Owner" ("id","name") VALUES ($1,$2) RETURNING "public"."Owner"."id"
INSERT INTO "public"."Car" ("id","name","ownerId") VALUES ($1,$2,$3) RETURNING "public"."Car"."id"
SELECT "public"."Car"."id", "public"."Car"."name", "public"."Car"."ownerId" FROM "public"."Car" WHERE "public"."Car"."id" = $1 LIMIT $2 OFFSET $3
COMMIT
```
`ownerId` is the id of the new owner which was created in the first call of the transaction.

**Comment by timsuchanek** [comment]
Closing as the question has been answered
